engine/language-server/src/main/scala/org/enso/languageserver/vcsmanager/Git.scala

```diff
@@ -45,17 +51,45 @@ private class Git extends VcsApi[BlockingIO] {
       if (!rootFile.exists()) {
         throw new FileNotFoundException("unable to find project repo: " + root)
       }
-      val repoLocation = root.resolve(DefaultGitRepoDir)
+
+      val repoLocation = root.resolve(gitDir)
       if (repoLocation.toFile.exists()) {
         throw new RepoExists()
       }
 
+      if (!repoLocation.getParent.toFile.exists()) {
+        if (!repoLocation.getParent.toFile.mkdirs()) {
+          throw new FileNotFoundException(
+            "unable to create project repository at " + repoLocation
+          )
+        }
+      }
+
       val jgit = JGit
         .init()
-        .setDirectory(rootFile)
+        .setGitDir(repoLocation.toFile)
+        .setDirectory(root.toFile)
         .setBare(false)
         .call()
 
+      ensoDataDirectory.foreach { path =>
+        // JGit **always** creates a .git file, gitdir path.
+        // The file may be confusing to the users and unnecessary for
+        // VCS since all commands include repo path explicitly.
+        val dotGit = root.resolve(".git").toFile
+        if (dotGit.exists()) {
+          dotGit.delete()
+        }
+
+        if (path.toFile.exists()) {
+          throw new EntryExistsException(path.toString)
```

Comment:
`ensoDataDir` (.enso) is used to store arbitrary information from various Enso systems, and most probably will be non-empty.